src/main/java/pl/nogacz/checkers/board/Board.java

```diff
@@ -136,6 +144,245 @@ public void readKeyboard(KeyEvent event) {
+                        save.write(whiteQueenCoords.get(i)+"\n");
+                    save.write(blackqueen_count + "\n");
+                    for(int i = 0; i < blackQueenCoords.size(); i++)
+                        save.write(blackQueenCoords.get(i)+"\n");                        
+                    save.close();
+                    System.out.println("You saved game!");
+                }
+                catch(IOException e){
+                    e.printStackTrace();
+                }   
+                alert.close();                
+            }           
+        }
+        if(event.getCode().equals(KeyCode.L)) {
+            Alert alert = new Alert(Alert.AlertType.NONE);
+            alert.setTitle("Load Game");
+            alert.setContentText("Game will be loaded from selected directory if you have saved before as 'JavaCheckedSavedGame.txt'.(Your Home Directory or Game Directory)");
+            ButtonType loadFromHome = new ButtonType("Home Directory");
+            ButtonType loadFromLocal = new ButtonType("Game Directory");  
+            alert.getButtonTypes().setAll(loadFromHome, loadFromLocal, ButtonType.CANCEL);
+            Optional<ButtonType> result = alert.showAndWait();
+            if (result.get() == loadFromLocal) {
+                try(BufferedReader load = new BufferedReader(new FileReader("JavaCheckersSavedGame.txt"))) {
+                    ArrayList<String> blackCoords = new ArrayList<String>();
+                    ArrayList<String> whiteCoords = new ArrayList<String>();
+                    ArrayList<String> blackQueenCoords = new ArrayList<String>();
+                    ArrayList<String> whiteQueenCoords = new ArrayList<String>();
+                    int white_count = Integer.parseInt(load.readLine());  
+                    for(int i = 0; i < white_count; i++)
+                        whiteCoords.add(load.readLine());
+                    int black_count = Integer.parseInt(load.readLine());     
+                    for(int i = 0; i < black_count; i++)
+                        blackCoords.add(load.readLine());     
+                    int whitequeen_count = Integer.parseInt(load.readLine());  
+                    for(int i = 0; i < whitequeen_count; i++)
+                        whiteQueenCoords.add(load.readLine());
+                    int blackqueen_count = Integer.parseInt(load.readLine());     
+                    for(int i = 0; i < blackqueen_count; i++)
+                        blackQueenCoords.add(load.readLine());                        
+                    load.close();          
+                    for(int x = 0; x <= 7; x++) {
+                        for(int y = 0; y <= 7; y++) {
+                            Coordinates coord = new Coordinates(x,y);
+                            Design.removePawn(coord);
+                            board.remove(coord);
+                        }
+                    }
+                    for(int i = 0; i < whiteCoords.size(); i++) {           

```

Comment:
Buradaki 4 for döngüsündeki kodlar hemen hemen aynı. Ortak bir metota bu kodları refactor et ve uygun parametrelerle o metodu çağır.